sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -675,3 +683,25 @@ def get_sql_chain(self, pipeline, set_user_pipeline=False):
             pipeline)
       chain.user_pipeline = pipeline
     return chain
+
+  def check_bucket_exists(self, bucket_name):
+    try:
+      from apitools.base.py.exceptions import HttpError
+    except:
+      raise ImportError('Could not import HttpError from apitools.')
```

Comment:
+1, if moving to the top, please follow the example: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/gcsio.py#L57

Or, see if the ImportError can be merged into the `unable to verify whether bucket {} exists` warning branch because you might not want the import error to fail the whole module.

I also think this function could be moved to the utils.py module: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/interactive/utils.py

Please add a docstring and typehints to this function so that we know what it does.
Things I'm not sure are:

- if the function is named `check_...`, shouldn't it return True/False like https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/gcsio.py#L461? But how will you differentiate "not exist" from "not able to check" by a bool value?
- if you rename the function to `assert_...` so that the function only asserts whether a bucket exists or not without a returning value, you may also need to add a `raise`:

```
except HttpError as e:
      if e.status_code == 404:
        _LOGGER.error('%s bucket does not exist!', bucket_name)
        raise
      else:
```

Otherwise, the error branch won't take effect.

Please also add a unit test to this function for different scenarios.

Reply:
Noted, thanks! I've moved the import to the top of utils.py and I've renamed the function to `assert_bucket_exists`.

In this case, I only raise a ValueError if the bucket cannot be found. In the scenario where the user may not be able to verify whether the bucket exists or not, I do not currently raise any exceptions so that the user could potentially still continue.

What are your thoughts on this approach?